src/test/deserialization_tests.cpp

```diff
@@ -85,6 +88,70 @@ class TensorflowGRPCPredict : public ::testing::TestWithParam<ovms::Precision> {
 class DeserializeTFTensorProto : public TensorflowGRPCPredict {};
 class DeserializeTFTensorProtoNegative : public TensorflowGRPCPredict {};
 
+class CAPIPredict : public ::testing::TestWithParam<ovms::Precision> {
+protected:
+    void SetUp() override {
+        auto precision = ovms::Precision::FP32;
+        tensorMap[tensorName] = std::make_shared<ovms::TensorInfo>(
+            tensorName,
+            precision,
+            shape_t{1, 3},
+            Layout{"NC"});
+        SetUpTensorProto(getPrecisionAsOVMSDataType(precision));
+    }
+    void SetUpTensorProto(DataType dataType) {
+        std::array<size_t, 2> shape{1, DUMMY_MODEL_INPUT_SIZE};
+        tensorCapi = std::make_unique<InferenceTensor>(OVMS_DATATYPE_FP32,
+            shape.data(),
+            shape.size());
+        bool createCopy{true};
+        const std::string data(1 * sizeof(float) * DUMMY_MODEL_INPUT_SIZE, '1');
+        auto status = tensorCapi->setBuffer(data.data(), data.size(), OVMS_BUFFERTYPE_CPU, std::nullopt, createCopy);
+        ASSERT_EQ(status, StatusCode::OK) << status.string();
+        const Buffer* buffer{nullptr};
+        buffer = tensorCapi->getBuffer();
+        ASSERT_NE(buffer, nullptr);
+    }
+    std::unique_ptr<InferenceTensor> tensorCapi;
+    const char* tensorName = DUMMY_MODEL_INPUT_NAME;
+    ovms::tensor_map_t tensorMap;
+    bool isPipeline = false;
+};
+
+class CAPIPredictRequest : public CAPIPredict {
+protected:
+    InferenceRequest request{"dummy", 1};
+    static const std::string DATA;
+    static constexpr std::array<size_t, DUMMY_MODEL_INPUT_SIZE> SHAPE{1, 10};
+    void SetUp() {
+        request.addInput(DUMMY_MODEL_INPUT_NAME,
+            OVMS_DATATYPE_FP32,
+            SHAPE.data(),
+            SHAPE.size());
+        request.setInputBuffer(DUMMY_MODEL_INPUT_NAME,
+            DATA.data(),
+            sizeof(float) * DUMMY_MODEL_INPUT_SIZE,
+            OVMS_BUFFERTYPE_CPU,
+            std::nullopt);
+    }
+    void TearDown() {
+    }
+};
+const std::string CAPIPredictRequest::DATA{1 * sizeof(float) * DUMMY_MODEL_INPUT_SIZE, '1'};
+// dispose connection to CAPI Predict
+TEST_F(CAPIPredictRequest, ShouldSuccessForSupportedPrecision) {
+    ov::Core ieCore;
+    std::shared_ptr<ov::Model> model = ieCore.read_model(std::filesystem::current_path().u8string() + "/src/test/dummy/1/dummy.xml");
+    ov::CompiledModel compiledModel = ieCore.compile_model(model, "CPU");
+    ov::InferRequest inferRequest = compiledModel.create_infer_request();
+    InputSink<ov::InferRequest&> inputSink(inferRequest);
+    auto status = deserializePredictRequest<ConcreteTensorProtoDeserializator>(request, tensorMap, inputSink, isPipeline);
+    EXPECT_TRUE(status.ok());
```

Comment:
Is ok() check enough to verify if precision deserialization works? You can set wrong precision and still the test will pass